app/controllers/application_controller.rb

```diff
@@ -67,9 +67,16 @@ def not_found
     raise ActionController::RoutingError, 'User Not Found'
   end
 
+  def user_identifying_params
+     params[:login] || params[:user_id] || params[:id]
+  end
+
   def find_user
-    login = params[:login] || params[:user_id] || params[:id]
-    @user = User.where(login: login).first || (current_user if %w[new favorites].include? action_name)
+    login = user_identifying_params
```

Comment:
@sudara I give up. I still get `Cognitive Complexity` failures from CodeClimate. Pls help :)

Reply:
Yeah, this is a `cognitively annoying` chunk of code. There are a few subtle things going on here that would probably be best broken out to be explicit, for example, when deleting users `params[:user_id]` is being manually passed to differentiate the user we want to delete and the `current_user`

This logic is old and needs to be updated. IMO the only responsible way to do that is to figure out when and where we use these different params and write the code to clearly communicate those things. It's not worth it to just please codeclimate.

For example, we are also saying "disregard the `login` param if the action is `new` or `favorites` — there's a reason for that but we'd have to lay that all out and maybe move that logic back into the appropriate controllers.

My bet is the end solution would be to have several methods that are explicitly named and called by the controllers that care. For example, `users_controller#destroy` could call `find_user_by_user_id_param` which would be located in its controller (if it actually makes sense to send that param still) and the favorites controller could have `set_user_to_current_user` or something...